### ISSUE TITLE
Confusion over use of term “wallet”

### ISSUE THREAD

**Issue by klmoney** [ethereum/meteor-dapp-wallet#95]
There are several places in the Wallet App UI where “wallet” is used in a general sense to refer to anything that can store ether, and then other places where it is used in a specific sense to mean contract-based wallet. “Wallet” is also used to refer to the Wallet App itself. This can cause confusion.

UI text that may be confusing now:

Where “wallet” is used in the general sense:
“Wallet Overview”
“WALLETS” tab
“Contract based wallets”
“Transfer between wallets” -- in Latest Transactions

Where “wallet” is used to mean “contract-based wallet”:
“Accounts can’t list incoming transactions, please create a wallet to store ether.”
“Wallets are contracts that hold and secure ether. They can have multiple accounts as owners and keep a full log of transactions.”
“ADD WALLET”

Where “wallet” is apparently referring to the Wallet App:
"Before using the Wallet you need to create at least one account with a strong password."
"You seem on a different chain, please restart the wallet!"

I wish to suggest that the term “wallet”, (or Wallet or WALLET) always be used in a general sense, to mean anything that can store ether (or other crypto tokens). So wallet can refer to any or all of the following: Account, Paper Wallet, Presale Wallet, a Wallet App, Multisig Wallet, etc.

If a specific type of wallet is meant, then it should be referred to by its specific name:
--Account: Externally Owned Account, as defined in the White Paper
--Contract Wallet (suggestion): contract-based wallet
--Simple Wallet: contract-based simple wallet
--Multisig Wallet: contract-based multisig wallet
--Paper Wallet, Presale Wallet, etc.
--Wallet App: to mean the wallet application. Otherwise use the full name be that Ethereum Wallet App or Mist Wallet App or Official Wallet App or whatever is decided as the name for this Wallet App.

tl;dr: I wish to suggest/request that contract-based wallets be given a name other than WALLET.

**Comment by alexvandesande** [comment]
I think you raise a great point and I agree. I will try to adopt more the term "contract-based wallet" or something like that.

There's a deeper complication here also and it is about the difference between accounts and contracts, and I'm not even sure how to tackle it yet..

**Comment by Souptacular** [comment]
I completely agree with you @klmoney. I think we should have better/new nouns and try to push for community feedback on it. The EAO accounts vs contract accounts thing should be solved in serenity when abstraction changes in Serenity described by @vbuterin take effect and there is only 1 account type.

**Comment by klmoney** [comment]
I guess the Wallet App UI really has the opportunity to establish standard terminology within the community for wallet related stuff. And standard terminology of course helps when it comes to describing/explaining/understanding how the app, and Ethereum wallets in general, work. There is a steady stream of wallet-related questions in the subreddit these days as I’m sure most have noticed.

**Comment by Souptacular** [comment]
@klmoney @alexvandesande In the next 1-2 months there will be both new Homestead documentation (I'm working on that) and new Mist related documentation Alex is working on. I propose that before the release of those documents we clarify the definitions and get community feedback.

**Comment by klmoney** [comment]
@Souptacular good idea.

**Comment by tymat** [comment]
I think it gets more confusing.  To me the following terms make sense.

**Wallet** - A container for account(s) (either a JSON file containing the encrypted private keys, an HD wallet, a paper wallet, presale wallet)

**Address** - an identifier that you send transactions to/from, a key on a ledger, a contract storage variable.  It can either be an Account or a Contract.

**Account** - An externally owned account (public and private keys)

**Comment by alexvandesande** [comment]
I'm thinking of using these definitions, what do you think:

**Accounts** are used to execute contracts, they live on your local machine and can hold ether or any ethereum based token.

**Smart accounts** are contract-based accounts: they live on the blockchain, can hold ether and tokens, can be owned by multiple accounts and keep a full log of all transactions

**Comment by tymat** [comment]
Should we use accounts or address?  In Ethereum programming we see references to address (i.e. variable types/function parameters are referred to as `address`  in most of the documentation) and accounts (i.e. web3.eth.accounts)

**Comment by frozeman** [comment]
I actually think keeping the term **wallet contract** vs **accounts** here. I think the whole issue is created because we weren't consistent enough with the naming in various places.

Introducing *smart-accounts*, which are technically just contracts, creates a mental separation between contracts and smart-accounts...

btw also normal accounts can hold tokens etc, the only thing they can't do is multisig, or other owner options.

Concerning the term `address`: I think this is not a clear distinction, as also accounts have *addresses*.

**Comment by tymat** [comment]
The yellow paper makes a distinction between externally owned accounts (i.e. they are pubkey/privkey derived) and contracts.

**Comment by rfikki** [comment]
"contract-address" and "account-address" would make this very understandable for both the lay person as well as programmers.

**Comment by tymat** [comment]
I think @rfikki just cracked the code.  I like it.

**Comment by frozeman** [comment]
Haha :)
Yes make sense

**Comment by klmoney** [comment]
Guys, I brought up this issue mainly because I have been writing up a little, unofficial, “for beginners” guide to using the ether management features of the Ethereum-Mist Wallet (seems like a popular name for now). And I found it very difficult to describe/explain while referring to contract-based wallets as WALLETS.

While waiting for the “official new name” to come out, I am using “Contract Wallet” in my guide. I thought it would only be courteous of me to mention this. Actually, right now, I kinda prefer the name “Smart Wallet”. But what I prefer is of course irrelevant as we will all have to go with the name the developers eventually put in the UI.

If anyone has the time to take a peek, the guide is here:
https://klmoney.wordpress.com/
And it is a work in progress of course :).

**Comment by frozeman** [comment]
I actually like the name "Mist Wallet". And we should probably just call it "Wallet contracts"

**Comment by klmoney** [comment]
@frozeman, I think Mist Wallet is a great name too. As for the new name for contract-based wallet, I look forward to seeing what you and @alexvandesande use in the next upgrade to the UI.

**Comment by prithwis** [comment]
@klmoney thanks for the beginners guide to the Ethereum Mist Wallet ... really appreciate your effort to make things easier for newbies like us.

**Comment by Souptacular** [comment]
What was the final outcome of this discussion? From what I am reading, it sounds like...

Contract-Address = A generally multi-sig based "wallet" that is really an Ethereum contract that holds/releases your Ether based on rules specified in the contract.
Account Address = Externally Owned Account (until Serenity comes) that can control Ether and be made a coinbase in your geth client.

**Comment by frozeman** [comment]
The finally outcome is an improvement to the texts in the wallet and the term "Wallet contract" and "Accounts"
